multiqc/modules/mirtrace/mirtrace.py

```diff
@@ -0,0 +1,381 @@
+            plot = self.mirtrace_rna_categories()
+        )
+
+        self.add_section (
+            name = 'Contamination Check',
+            anchor = 'mirtrace_contamination_check',
+            plot = self.mirtrace_contamination_check()
+        )
+
+        self.add_section (
+            name = 'miRNA Complexity',
+            anchor = 'mirtrace_complexity',
+            plot = self.mirtrace_complexity_plot()
+        )
+
+    # Parse a miRTrace results.json file
+    def parse_summary(self, f):
+
+        try:
+            cdict = json.loads(f['f'])
+        except ValueError as e:
+            raise e
+
+        if 'results' in cdict.keys():
+            for record in cdict['results']:
+                s_name = self.clean_s_name(record['verbosename'], f['root'])
+                parsed_data = {}
+                parsed_data['filename'] = record['filename']
+                parsed_data['reads_total'] = record['stats']['allSeqsCount']
+                parsed_data['adapter_removed_length_ok'] = record['stats']['statsQC'][4]
+                parsed_data['adapter_not_detected'] = record['stats']['statsQC'][3]
+                parsed_data['length_shorter_than_18'] = record['stats']['statsQC'][2]
+                parsed_data['low_complexity'] = record['stats']['statsQC'][1]
+                parsed_data['low_phred'] = record['stats']['statsQC'][0]
+                parsed_data['reads_mirna'] = record['stats']['statsRNAType'][0]
+                parsed_data['reads_rrna'] = record['stats']['statsRNAType'][1]
+                parsed_data['reads_trna'] = record['stats']['statsRNAType'][2]
+                parsed_data['reads_artifact'] = record['stats']['statsRNAType'][3]
+                parsed_data['reads_unknown'] = record['stats']['statsRNAType'][4]
+                if s_name in self.summary_data:
+                    s_name = s_name + '_' + record['filename']
+                self.add_data_source(f, s_name)
+                self.summary_data[s_name] = parsed_data
+
+
+    # Parse a miRTrace mirtrace-stats-length.tsv file
+    def parse_length(self, f):
+        try:
+            with open(f['fn'], 'rb') as file:
+                reader = csv.reader(file, delimiter="\t")
+                header = reader.next()
+                body = {}
+                for row in reader:
+                    body[row[0]]=row[1:len(row)]
+        except ValueError as e:
+            raise e
+
+        if header[0] == 'LENGTH':
+            for record in header[1:len(header)]:
+                s_name = self.clean_s_name(record, f['root'])
+                parsed_data = {}
+                idx = header[1:len(header)].index(record)
+                for length in body:
+                    parsed_data[length] = int(body[length][idx])
+                if s_name in self.length_data:
+                    s_name = s_name + '_' + ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in range(6))
```

Comment:
Why do you need to do this? When would the `s_name` be the same? Usually in these cases we overwrite the previous data and flag a debug log message:
https://github.com/ewels/MultiQC/blob/dd262ff4bbc5926b2ace4135f4a7ed36c021085f/multiqc/modules/salmon/salmon.py#L45-L46

Reply:
Hi, I was thinking about the situation that multiple input fastq files were anlayzed for one sample. It will not happen based on our routine, but maybe someone else will run into such situation.